# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/css/css-inline/baseline-source/baseline-source-first-textarea-003.tentative.html
# Commit: 2187935f6d78
# Full Hash: 2187935f6d78c22ceeb7e5e92e919aeb42820b95
# Author: David Shin <dshin@mozilla.com>
# Date: 2023-08-16 03:43:43
# Regressor Bug: 1846982
# File Overlap Count: 1
# Description:
#   Bug 1846982: Baseline calculation for textarea. r=emilio
#   
#   Tests are added as tentative, because the behaviour when there's no content,
#   or when there's placeholder text, seem very differnt for each browser.
#   
# ==============================================================================

diff -r 2d475d0406cd -r 2187935f6d78 testing/web-platform/tests/css/css-inline/baseline-source/baseline-source-first-textarea-003.tentative.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/css/css-inline/baseline-source/baseline-source-first-textarea-003.tentative.html	Tue Aug 15 13:30:17 2023 +0000
@@ -0,0 +1,70 @@
+<!DOCTYPE html>
+<link rel="help" href="https://drafts.csswg.org/css-inline-3/#baseline-source">
+<link rel="author" title="David Shin" href="mailto:dshin@mozilla.com">
+<link rel="author" title="Mozilla" href="http://www.mozilla.org/">
+<meta name="assert" content="<textarea> with baseline-source: first generates sensible baselines.">
+<link rel="stylesheet" href="/fonts/ahem.css">
+<style>
+.target {
+  inline-size: 200px;
+  padding: 10px;
+  border: solid 3px;
+  position: relative;
+  line-height: 0;
+  font-size: 20px;
+  writing-mode: vertical-lr;
+}
+
+.inner {
+  border: solid 5px;
+  padding: 10px;
+  baseline-source: first;
+  font-size: 30px;
+}
+
+span {
+  display: inline-block;
+  width: 1em;
+  height: 1em;
+  outline: solid cyan 3px;
+  outline-offset: -3px;
+}
+
+textarea {
+  border: solid 5px;
+  width: 60px;
+  height: 60px;
+  vertical-align: baseline;
+  font-family: Ahem;
+  line-height: 1;
+}
+
+</style>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/check-layout-th.js"></script>
+<body onload="checkLayout('.target > *')">
+<div class="target">
+  <span data-offset-x="31"></span>
+  <textarea data-offset-x="11" class="inner"></textarea>
+</div>
+<div class="target">
+  <span data-offset-x="31"></span>
+  <textarea data-offset-x="11" class="inner">X X X X X</textarea>
+</div>
+<div class="target">
+  <span data-offset-x="31"></span>
+  <textarea data-offset-x="11" class="inner" placeholder="X X X X X"></textarea>
+</div>
+<div class="target">
+  <span data-offset-x="66"></span>
+  <textarea data-offset-x="11" class="inner" style="font-size: 100px;"></textarea>
+</div>
+<div class="target">
+  <span data-offset-x="66"></span>
+  <textarea data-offset-x="11" class="inner" style="font-size: 100px;">X X X X X</textarea>
+</div>
+<div class="target">
+  <span data-offset-x="66"></span>
+  <textarea data-offset-x="11" class="inner" style="font-size: 100px;" placeholder="X X X X X"></textarea>
+</div>